config/identifiable.go

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"errors"
+	"strings"
+)
+
+// typeAndNameSeparator is the separator that is used between type and name in type/name composite keys.
+const typeAndNameSeparator = "/"
+
+// ID represents the identity for a component. It combines two values:
+// * type - the Type of the component.
+// * name - the name of that component.
+// The component ID (combination type + name) is unique for a given component.Kind.
+type ID struct {
```

Comment:
This becomes `config.ID` elsewhere, which is confusing because it appears we are talking about configuration id. It is a component id, not a config id, so maybe name it `config.ComponentID`:
```suggestion
type ComponentID struct {
```

Reply:
Sounds good.